plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -310,7 +310,6 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
         dropTableData(table.io(), table.operations().current());
-        deleteTableDirectory(session, schemaTableName, hdfsEnvironment, new Path(table.location()));
```

Comment:
It's not desirable.

We don't want users to have to run remove_orphan_files prior to DROP TABLE to ensure no PII data is left behind on the storage.